client/src/tasks/network_scan.rs

```diff
@@ -0,0 +1,85 @@
+use std::net::{Ipv4Addr, SocketAddr, ToSocketAddrs};
+use std::ptr::addr_of;
+use std::sync::{Arc, Mutex};
+use std::time::{Duration, Instant};
+use anyhow::Error;
+use ipnet::Ipv4Net;
+use local_ip_address::local_ip;
+use futures::{stream, StreamExt};
+use serde_json::to_string;
+use tokio::net::TcpStream;
+use crate::models::{
+    NetworkHost,
+    Port,
+};
+
+pub const COMMON_PORTS: &[u16] = &[21, 22, 23, 25, 53, 80, 443, 8000, 8080];
+
+
+async fn scan_port(ip: &String, port: &u16) -> Port {
+    let timeout = Duration::from_millis(25);
+    let socket_address: Vec<SocketAddr> = format!("{}:{}", ip, port)
+        .to_socket_addrs()
+        .expect("port scanner: Creating socket address")
+        .collect();
+
+    let is_open = matches!(
+        tokio::time::timeout(timeout, TcpStream::connect(&socket_address[0])).await,
+        Ok(Ok(_))
+    );
+    Port {
+        port: *port,
+        is_open,
+    }
+}
+
+
+async fn scan_ports(ip: Ipv4Addr, ports: &Vec<u16>) -> NetworkHost {
+    let mut host = NetworkHost {
+        ip: ip.to_string(),
+        open_ports: vec![],
+    };
+    let ports: Vec<Port> = stream::iter(ports.into_iter())
```

Comment:
Nice to see that ports appear to be scanned for a host asynchronously using futures!